paddle/framework/channel_test.cc

```diff
@@ -100,6 +100,88 @@ TEST(Channel, SimpleUnbufferedChannelTest) {
   delete ch;
 }
 
+// This tests that closing an unbuffered channel also unblocks
+//  unblocks any receivers waiting for senders
+TEST(Channel, UnbufferedChannelCloseUnblocksReceiversTest) {
+  auto ch = MakeChannel<int>(0);
+  size_t num_threads = 5;
+  std::thread t[num_threads];
+  bool thread_ended[num_threads];
+
+  // Launches threads that try to read and are blocked becausew of no writers
+  for (size_t i = 0; i < num_threads; i++) {
+    thread_ended[i] = false;
+    t[i] = std::thread(
+        [&](bool *p) {
+          int data;
+          ch->Receive(&data);
```

Comment:
There are two possibilities for the return of `Receive`:
1. the `Channel` was closed.
2. the `Send` of this `Channel` was called.

I think we should separate them. When the `Channel` was closed, the point of `data` should be `nullptr`. Otherwise, it should be not `nullptr`.

Reply:
Do you mean the data pointer that is passed to `Receive`, that should be null.